packages/modeling/src/primitives/polyhedron.js

```diff
@@ -32,13 +35,25 @@ const polyhedron = (options) => {
   if (faces.length < 1) {
     throw new Error('one or more faces are required')
   }
+  if (colors) {
+    if (!Array.isArray(colors)) {
+      throw new Error('colors must be an array')
+    }
+    if (colors.length !== faces.length) colors = undefined
+  }
 
   // invert the faces if orientation is inwards, as all internals expect outwarding facing polygons
   if (orientation !== 'outward') {
     faces.forEach((face) => face.reverse())
   }
 
   let polygons = faces.map((face) => poly3.fromPoints(face.map((idx) => points[idx])))
+
+  if (colors) {
+    // add color to each polygon
+    polygons.forEach((polygon, idx) => { if (colors[idx]) polygon.color = colors[idx] })
```

Comment:
hmm this could get very expensive: 
- the faces are mapped over (1 iteration)
- then the polygons are iterated over again to add color
Perhaps adding poly3.fromPointsAndColor makes sense ?
I think it is not essential to do this right now, but at least we should have TODO note in the code to add this later

Reply:
Good catch. I merged to two loops.
Done